### DISCUSSION TITLE
Data Kiosk graphql "start/end date" does not support "time"

### DISCUSSION THREAD

**Discussion by DearRude** [amzn/selling-partner-api-models#4214] [General] [answered]
I have a valid rfc3339 time input for "start/end date" like the following: `2024-10-08T10:04:05Z`
However it returns data validation error unless `time` part is omitted like the following: `2024-10-08`

**Comment by mafge** [comment]
Hi @DearRude,
thanks for reaching out!
Using date only is the correct way. I admit that referencing RFC3339 in the schema description is misleading and it could be difficult to  understand that only the date portion of it should be set. Hope that helps!
Best,
Marc
Selling Partner Developer Services

**Comment by chrisskhoury** [comment, marked as answer]
Hello @DearRude,

The [Schema Explorer](https://sellercentral.amazon.com/datakiosk-schema-explorer?schema=analytics_salesAndTraffic_2024_04_24) should help clarify this. If you navigate to the right panel and select the Date model, you will find the date/time format specified. In this case, it is RFC 3339.

RFC 3339 specifies a format for representing date and time, but it also allows for just the date part. This is highlighted in the example section.

Currently, there are no datasets that provide data at an hour or minute granularity. The format remains RFC3339 but without the hours, minutes, and seconds values. You can see an example in the Schema Explorer:

> An RFC-3339 compliant Full Date Scalar. Example: "2023-11-15"

This format is used in anticipation of future datasets that might include hour or minute granularity.

I understand this can be a bit confusing, and we will work on highlighting this more clearly in the example section.

Thanks for the feedback and please keep them coming so that we can improve your experience :)

All the best,

Chris

**Reply by DearRude** [reply]
Thank you for the response.

I am also looking for a solution to get near real-time dashboard of "`units/orders sold`". However, I cannot get _today_'s data using Kiosk api. It returns empty query for this response.
Am I configuring something wrong or Kiosk is not intended for this purpose?

**Reply by chrisskhoury** [reply]
The data in SKU Economics Dataset is updated every ~24 hours. The data in Vendor Analytics  and Sales and Traffic is updated every ~34-36 hours. At the moment, with the current dataset you can get yesterday's data.

If you are trying to get the real time data for vendor analytics, we do also have the [GET_VENDOR_REAL_TIME_SALES_REPORT](https://developer-docs.amazon.com/sp-api/docs/report-type-values-analytics#rapid-retail-analytics-sales-report) that provides this information aggregated to an hourly granularity. 

But, I would still recommend adopting Kiosk, because it is the new tool for upcoming metrics and new datasets are always being pushed. It is a fairly easy API to implement and having it ready would give you access to additional data moving forward.

**Reply by DearRude** [reply]
We are currently using `GET_FLAT_FILE_ALL_ORDERS_DATA_BY_LAST_UPDATE_GENERAL`  report and culculate sold units manually. I was wondering if Kiosk (or generally any api) provides a better alternative that can mimic near real-time results.
